From 682263d558c6bd5ceb7e729519aa10e3a8668071 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 27 Mar 2014 19:07:43 -0400
Subject: [tools] Revert: turbostat: servers do not support uncore power register

Message-id: <1395947264-29338-2-git-send-email-prarit@redhat.com>
Patchwork-id: 78662
O-Subject: [RHEL7.1 PATCH BZ 1081657 1/2] Revert "[tools] turbostat: servers do not support uncore power register"
Bugzilla: 1081657
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1081657

This reverts commit 73d9fc596cbc3943a78524a83103163b803c4996.

This is being reverted in favour of a better upstream commit that was
applied after this patch was submitted upstream.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/x86/turbostat/turbostat.c b/tools/power/x86/turbostat/turbostat.c
index ff505c2..9d77f13 100644
--- a/tools/power/x86/turbostat/turbostat.c
+++ b/tools/power/x86/turbostat/turbostat.c
@@ -219,7 +219,7 @@ int cpu_migrate(int cpu)
   return 0;
 }
 
-int _get_msr(int cpu, off_t offset, unsigned long long *msr, int warn)
+int get_msr(int cpu, off_t offset, unsigned long long *msr)
 {
  ssize_t retval;
  char pathname[32];
@@ -234,25 +234,13 @@ int _get_msr(int cpu, off_t offset, unsigned long long *msr, int warn)
  close(fd);
 
  if (retval != sizeof *msr) {
-  if (warn)
-   fprintf(stderr, "%s offset 0x%zx read failed\n",
-    pathname, offset);
+  fprintf(stderr, "%s offset 0x%zx read failed\n", pathname, offset);
   return -1;
  }
 
  return 0;
 }
 
-int get_msr(int cpu, off_t offset, unsigned long long *msr)
-{
- return _get_msr(cpu, offset, msr, 1);
-}
-
-int get_msr_nowarn(int cpu, off_t offset, unsigned long long *msr)
-{
- return _get_msr(cpu, offset, msr, 0);
-}
-
 void print_header(void)
 {
  if (show_pkg)
@@ -965,10 +953,9 @@ int get_counters(struct thread_data *t, struct core_data *c, struct pkg_data *p)
   p->energy_dram = msr & 0xFFFFFFFF;
  }
  if (do_rapl & RAPL_GFX) {
-  if (get_msr_nowarn(cpu, MSR_PP1_ENERGY_STATUS, &msr))
-   do_rapl &= ~RAPL_GFX; /* unsupported on servers */
-  else
-   p->energy_gfx = msr & 0xFFFFFFFF;
+  if (get_msr(cpu, MSR_PP1_ENERGY_STATUS, &msr))
+   return -16;
+  p->energy_gfx = msr & 0xFFFFFFFF;
  }
  if (do_rapl & RAPL_PKG_PERF_STATUS) {
   if (get_msr(cpu, MSR_PKG_PERF_STATUS, &msr))
-- 
1.7.1